src/Microsoft.OData.Core/UriParser/Binders/SelectPathSegmentTokenBinder.cs

```diff
@@ -155,7 +155,12 @@ internal static bool TryBindAsOperation(PathSegmentToken pathToken, IEdmModel mo
                     NonSystemToken nonSystemToken = pathToken as NonSystemToken;
                     if (nonSystemToken != null && nonSystemToken.NamedValues != null)
                     {
-                        parameterNames = nonSystemToken.NamedValues.Select(s => s.Name).ToList();
+                        parameterNames = new List<string>();
+
+                        foreach(NamedValue namedValue in nonSystemToken.NamedValues)

```

Comment:
The linq query on the right could be faster in most cases as the list may need to resize during addition.

Reply:
But it does 2 iteration right. 1 for select and another for T0List

Reply:
```suggestion
                        foreach (NamedValue namedValue in nonSystemToken.NamedValues)

```

Reply:
@Sreejithpin  does it really do 2 iterations? I think `.Select()` is lazily evaluated when `ToList()` is called, and probably leads to just one iteration? I'd be curious to see the benchmarks for this particular loop compared to the code being replaced.